File: 30 days of code/__ C#/Day 16 - Exceptions (String to Integer).cs
/**
 *              EXCEPTIONS (STRING TO INTEGER) | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 16
 */

class Program
{
    //static TextReader read = Console.In;
    static StreamReader read = new StreamReader("input.txt");

    static void Main(string[] args)
    {
        string input = read.ReadLine();
        read.Close();

        try
        {
            int n = Int32.Parse(input);
            Console.WriteLine(n);
        }
        catch (FormatException fe)
        {
            Console.WriteLine("Bad String");
        }
    }
}

File: 30 days of code/__ C++/Day 16 - Exceptions (String to Integer).cpp
/**
*              EXCEPTIONS (STRING TO INTEGER) | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 16
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

ifstream file("input.txt");
istream& f = file;

void main() {
	string input; getline(f, input);

	try {
		int n = stoi(input);
		cout << n << endl;
	}
	catch (...) {
		cout << "Bad String" << endl;
	}
}